index.src.html

```diff
@@ -2958,80 +3017,232 @@
+        3. Assign all remaining attributes of |frame| (
+            {{VideoFrame/codedWidth}}, {{VideoFrame/codedHeight}}, etc.) to
+            those of the same name in |clone|.
     2. Return |clone|.
 
-Plane Interface {#plane-interface}
-----------------------------------
-A {{Plane}} is solely constructed by its {{VideoFrame}}. During construction,
-    the User Agent may use knowledge of the frame’s {{PixelFormat}} to add
-    padding to the {{Plane}} to improve memory alignment.
-
-A {{Plane}} cannot be used after the {{VideoFrame}} is destroyed. A new
-    {{VideoFrame}} can be assembled from existing {{Plane}}s, and the new
-    {{VideoFrame}} will remain valid when the original is destroyed. This makes
-    it possible to efficiently add an alpha plane to an existing
-    {{VideoFrame}}.
+: <dfn for=VideoFrame>Parse VideoFrameCopyToOptions</dfn> (with |options|)
+:: 1. Let |parsedRect| be the result of running the [=VideoFrame/Parse CopyTo
+        Rect=] with |options|.
+    2. If |parsedRect| is an exception, return |parsedRect|.
+    3. Let |parsedOptions| be the result of running the [=VideoFrame/Compute
+        Layout and Allocation Size=] algorithm with |parsedRect| and |options|.
+    6. Return |parsedOptions|.
+
+: <dfn for=VideoFrame>Parse CopyTo Rect</dfn> (with |options|)
+:: 1. Let |sourceRect| be the be the result of performing the getter steps for
+        {{VideoFrame/visibleRect}}.
+    2. If {{VideoFrameCopyToOptions/rect}} [=map/exists=] in |options|:
+        1. Let |optRect| be {{VideoFrameCopyToOptions/rect}}:
+        2. If either of |optRect|.{{VideoFrameRect/width}} or
+            {{VideoFrameRect/height}} is `0`, return a {{TypeError}}.
+        3. If the sum of |optRect|.{{VideoFrameRect/left}} and
+            |optRect|.{{VideoFrameRect/width}} is greater than
+            {{VideoFrame/[[coded width]]}}, return a {{TypeError}}.
+        4. If the sum of |optRect|.{{VideoFrameRect/top}} and
+            |optRect|.{{VideoFrameRect/height}} is greater than
+            {{VideoFrame/[[coded height]]}}, return a {{TypeError}}.
+        5. Assign |optRect| to |sourceRect|.
+    3. Let |planeIndex| be `0`.
+    4. Let |numPlanes| be the number of planes as defined by
+        {{VideoFrame/[[format]]}}
+    5. While |planeIndex| is less than |numPlanes|:
+
+        NOTE: The following steps validate |rect| is sample-aligned for this
+            frame's {{VideoFrame/[[format]]}}.
+
+        1. Let |plane| be the Plane identified by |planeIndex| as defined by
+            {{VideoFrame/[[format]]}}.
+        2. Let |sampleWidth| be the horizontal pixel size of each subsample
+            for |plane|.
+        3. Let |sampleHeight| be the vertical pixel size of each subsample
+            for |plane|.
+        4. If |sourceRect|.{{VideoFrameRect/left}} and
+            |sourceRect|.{{VideoFrameRect/width}} cannot be evenly divided by
+            |sampleWidth|, throw a {{TypeError}}.
```

Comment:
I see this formulation "is not a multiple of" more often for alignment stuff. I checked HTML and that seems to be the case for imagedata stuff.

Reply:
Fixed here and below.